tests/executables/kafka_topic_gui.py

```diff
@@ -0,0 +1,320 @@
+        with VerticalScroll():
+            for instruction in self.instructions:
+                yield instruction
+        with Horizontal():
+            for option_callback in self.option_callbacks:
+                yield Button(option_callback, id=option_callback)
+
+    def on_button_pressed(self, event: Button.Pressed):
+        event.stop()
+
+        if (b_id := event.button.id) is not None and (
+            callback := self.option_callbacks.get(b_id)
+        ) is not None:
+            callback()
+
+        self.app.pop_screen()
+
+
+class KafkaTopicManager(Screen):
+    DEFAULT_CSS = """
+    KafkaTopicButtons {
+        height: auto;
+        padding: 1;
+    }
+
+    VerticalScroll {
+        height: auto;
+        max-height: 10;
+        padding: 1;
+    }
+
+    SelectionList {
+        padding: 1;
+        border: solid $accent;
+    }
+
+    KafkaTopicInput {
+        height: auto;
+        border: solid $accent;
+    }
+    """
+
+    def __init__(self, kafka_server: str) -> None:
+        from functools import partial
+
+        from confluent_kafka.admin import AdminClient
+
+        super().__init__(name="Kafka Topic Control Screen")
+
+        self.admin = AdminClient(conf={"bootstrap.servers": kafka_server})
+        self.action_map = {
+            'refresh': self.refresh_topics,
+            'delete': self.delete_selected,
+            'select-all': partial(self.select_or_deselect_all, True),
+            'deselect-all': partial(self.select_or_deselect_all, False),
+        }
+        self.topics = SelectionList()
+        self.topic_container = VerticalScroll(self.topics)
+        self.topic_container.border_title = f"Topics at {kafka_server}"
+
+    @work(exclusive=True)
+    async def refresh_topics(self) -> None:
+        import asyncio
+
+        from textual.widgets.option_list import OptionDoesNotExist
+
+        def compose_selection(topic_id: str) -> Selection:
+            try:
+                return self.topics.get_option(topic_id)
+            except OptionDoesNotExist:
+                return Selection(topic_id, topic_id, False, id=topic_id)
+
+        all_topics = [
+            topic
+            for topic in self.admin.list_topics().topics
+            if not topic.startswith('_')
+        ]
+
+        all_topics.sort()
+        self._refresh_selection_list()
+        self.topics.add_options([compose_selection(topic) for topic in all_topics])
+        await asyncio.sleep(1)  # At maximum once per second.
+
+    def _refresh_selection_list(self):
+        """``SelectionList`` creating helper.
+
+        ``SelectionList`` causes an error at the exit of the application
+        """
+        self.topics.remove()
+        self.topics = SelectionList()
+        self.topic_container.mount(self.topics)
+
+    @work(exclusive=True)
+    async def delete_confirmed(self, selected: list[str]) -> None:
+        import asyncio
+
+        selected = self.topics.selected
+        self.admin.delete_topics(selected)
```

Comment:
Doesn't the `delete_topics` method return some kind of future that indicates when the topics are deleted?
If it does then I think it would be better to await that future instead of sleeping for a fixed time.

Reply:
Yes, that makes more sense. I'll fix it that way.

Reply:
I fixed it so that the users will see this waiting screen while kafka topics are deleted, I add artificially added async sleep line so that you can see the waiting screen but it almost never shows up since topics are automatically deleted in the testing environments.

![wait_screen](https://github.com/scipp/beamlime/assets/17974113/541847cc-1f35-43cb-afe5-fd190a18ac07)